src/main/java/com/stuypulse/robot/subsystems/ColorSensor.java

```diff
@@ -5,24 +5,95 @@
 
 package com.stuypulse.robot.subsystems;
 
+import com.revrobotics.ColorMatch;
+import com.revrobotics.ColorMatchResult;
+import com.revrobotics.ColorSensorV3;
+
+import com.stuypulse.robot.Constants.ColorSensorSettings;
+import com.stuypulse.robot.Constants.ColorSensorSettings.BallColor;
+import com.stuypulse.robot.Constants.Ports;
+
+import edu.wpi.first.wpilibj.DriverStation.Alliance;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+
+import edu.wpi.first.wpilibj.util.Color;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
 
 /*-
- * Detects what color ball is in the [IDK]
+ * Detects what color ball is in the Conveyor.
  *
  * Contains:
- *      - TBD
+ *      - getCurrentBall()
+ *         - Gets current ball seen by subsystem.
+ *         - Returns CurrentBall enum
+ *      - gapHasAllianceBall()
+ *         - Checks if there is a ball and the ball is the correct alliance color
+ *         - Returns false if no ball or wrong Alliance color
+ *      - isBallPresent()
+ *         - Checks if there is a ball present
  *
- * @author [TBD]
+ * @author Vincent Wang
  */
 public class ColorSensor extends SubsystemBase {
+    public enum CurrentBall {
+        RED_BALL(Alliance.Red),
+        BLUE_BALL(Alliance.Blue),
+        NO_BALL(Alliance.Invalid);
```

Comment:
it might make more sense to make this null (and change around the .equals or use ==) but based on the if statement on gapHasAllianceColor it should never compare with NO_BALL (and its allianceColor, so it doesnt matter what the value is)